sdk/test-utils/recorder/src/recorder.ts

```diff
@@ -53,19 +58,58 @@ export class Recorder {
   private assetsJson?: string;
   private variables: Record<string, string>;
 
-  constructor(private testContext?: Test | undefined) {
+  constructor(
+    private testContext?:
+      | MochaTest
+      | VitestContext
+      | undefined,
+  ) {
+    if (!this.testContext) {
+      throw new Error(
+        "Unable to determine the recording file path, testContext provided is not defined.",
+      );
+    }
+
     logger.info(`[Recorder#constructor] Creating a recorder instance in ${getTestMode()} mode`);
     if (isRecordMode() || isPlaybackMode()) {
       if (this.testContext) {
-        this.sessionFile = sessionFilePath(this.testContext);
+        let context:
+          | MochaTest
+          | {
+              suiteTitle: string;
+              testTitle: string;
+            };
+        if (this.testContext instanceof MochaTest) {
+          if (!this.testContext.parent) {
+            throw new RecorderError(
+              `Test ${this.testContext.title} is not inside a describe block, so a file path for its recording could not be generated. Please place the test inside a describe block.`,
+            );
+          }
+          context = {
+            suiteTitle: this.testContext.parent.fullTitle(),
+            testTitle: this.testContext.title,
+          };
+        } else {
+          if (!this.testContext.currentTestName) {
+            throw new RecorderError(`Unable to determine the recording file path. Unexpected empty Vitest context`);
+          }
+          // vitest format:
+          //   ""fileName.ts > suite title [> nested suite title ]* > test title""
+          const parts = this.testContext.currentTestName.split(" > ");
+          if (parts.length < 3) {
+            throw new RecorderError(`Unable to determine the recording file path. Unexpected Vitest currentTestname`);
+          }
+          context = {
+            suiteTitle: parts.slice(1, parts.length - 1).join("_").trim(),
+            testTitle: parts[parts.length - 1].trim(),
```

Comment:
could pull this out and make a unit test out of it to simplify the constructor :)